src/Uno.UWP/System/Display/DisplayRequest.cs

```diff
@@ -0,0 +1,50 @@
+﻿#if __ANDROID__ || __IOS__
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Windows.System.Display
+{
+	public partial class DisplayRequest
+	{
+		private static object _syncLock = new object();
+		private static int _globalRequestCount = 0;
+
+		private int _instanceRequestCount = 0;
+
+		public void RequestActive()
+		{
+			lock (_syncLock)
+			{
+				_instanceRequestCount++;
+				_globalRequestCount++;
+				if (_globalRequestCount == 1)
```

Comment:
I think there's a way to implement this in a lockless way using `Interlocked.Increment` / `Interlocked.Decrement`

Reply:
Not sure how to approach this though - as I need to increment/decrement two variables, but the most important thing is that the `ActivateScreenLock` and `DeactivateScreenLock` must run only when the `_globalRequestCount` has an appropriate value. I think I can make sure to safely increment with `Interlocked`, but that won't solve the problem when I check `_globalRequestCount` value (in `if`), enter the block, and at that moment another thread changes the `_globalRequestCount` again, I may for example then deactivate the screen lock although another thread just activated it. Do you have an idea how to resolve this problem?

Reply:
I think the `DisplayRequest` class will not commonly be used in high-perf scenarios, most apps will just use it once per launch, for example to keep screen on when playing video, so the cost of the `lock` feels manageable and it is easy to show that the code works correctly as compared to a more complex solution